### PR TITLE
Update to latest go-amqp

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Includes updates due to API breaking changes.
Cleaned up various linter bugs.
Removed some dead code.

### CODE REVIEW

File: namespace.go [hunk 1]
```diff
@@ -282,7 +282,7 @@ func (ns *Namespace) negotiateClaim(ctx context.Context, client *amqp.Client, en
 						tab.For(refreshCtx).Error(err)
 						select {
 						case <-refreshCtx.Done():
-							break
+							return
```
Comment (richardpark-msft) [inline namespace.go:285]: Hm...makes sense. I guess the claim negotiation would just never end?

File: subscription.go [hunk 2]
```diff
@@ -36,12 +36,11 @@ type (
...
-		receiver          *Receiver
-		receiverMu        sync.Mutex
-		receiveMode       ReceiveMode
-		requiredSessionID *string
-		prefetchCount     *uint32
+		Topic         *Topic
+		receiver      *Receiver
+		receiverMu    sync.Mutex
+		receiveMode   ReceiveMode
+		prefetchCount *uint32
```
Comment (richardpark-msft) [inline subscription.go:43]: requiredSessionID was just unused?

**Review by richardpark-msft — APPROVED** [review]
Seems fine to me.

File: subscription.go (hunk 2 above)
Comment (jhendrixMSFT) [reply subscription.go:43]: Yeah

File: namespace.go (hunk 1 above)
Comment (jhendrixMSFT) [reply namespace.go:285]: Yep, stuck in the for loop.

**Pull request merged**